packages/@aws-cdk/aws-elasticsearch/lib/domain.ts

```diff
@@ -0,0 +1,611 @@
+   *
+   * @default maximum over a minute
+   */
+  metricKMSKeyError(clientId: string, props?: MetricOptions): Metric;
+
+  /**
+   * Metric for KMS key being inaccessible.
+   *
+   * @default maximum over a minute
+   */
+  metricKMSKeyInaccessible(clientId: string, props?: MetricOptions): Metric;
+
+  /**
+   * Metric for number of searchable documents.
+   *
+   * @default maximum over a minute
+   */
+  metricSearchableDocuments(clientId: string, props?: MetricOptions): Metric;
+
+  /**
+   * Metric for search latency.
+   *
+   * @default maximum over a minute
+   */
+  metricSearchLatency(clientId: string, props?: MetricOptions): Metric;
+
+  /**
+   * Metric for indexing latency.
+   *
+   * @default maximum over a minute
+   */
+  metricIndexingLatency(clientId: string, props?: MetricOptions): Metric;
+}
+export class Domain extends cdk.Resource implements IDomain {
+  /**
+   * @attribute
+   */
+  public readonly domainArn: string;
+
+  /**
+   * @attribute
+   */
+  public readonly domainName: string;
+
+  /**
+   * @attribute
+   */
+  public readonly domainEndpoint: string;
+
+
+  private readonly domain: CfnDomain;
+
+  private readonly slowSearchLogGroup?: logs.LogGroup;
+
+  private readonly slowIndexLogGroup?: logs.LogGroup;
+
+  private readonly appLogGroup?: logs.LogGroup;
+
+  constructor(scope: cdk.Construct, id: string, props: DomainProps) {
+    super(scope, id, {
+      physicalName: props.domainName,
+    });
+
+    // Setup logging
+    const logGroups: logs.LogGroup[] = [];
+
+    if (props.logPublishingOptions?.slowSearchLogGroup) {
+      this.slowSearchLogGroup = props.logPublishingOptions.slowSearchLogGroup;
+      logGroups.push(this.slowSearchLogGroup);
+    } else if (props.logPublishingOptions?.slowSearchLogEnabed) {
+      this.slowSearchLogGroup = new logs.LogGroup(this, 'SlowSearchLogs', {
+        logGroupName: `elasticsearch/domains/${props.domainName ?? id}/slow-search-logs`,
+        retention: logs.RetentionDays.ONE_MONTH,
+      });
+      logGroups.push(this.slowSearchLogGroup);
+    }
+
+    if (props.logPublishingOptions?.slowIndexLogGroup) {
+      this.slowIndexLogGroup = props.logPublishingOptions.slowIndexLogGroup;
+      logGroups.push(this.slowIndexLogGroup);
+    } else if (props.logPublishingOptions?.slowIndexLogEnabed) {
+      this.slowIndexLogGroup = new logs.LogGroup(this, 'SlowIndexLogs', {
+        logGroupName: `elasticsearch/domains/${props.domainName ?? id}/slow-index-logs`,
+        retention: logs.RetentionDays.ONE_MONTH,
+      });
+      logGroups.push(this.slowIndexLogGroup);
+    }
+
+    if (props.logPublishingOptions?.appLogGroup) {
+      this.appLogGroup = props.logPublishingOptions.appLogGroup;
+      logGroups.push(this.appLogGroup);
+    } else if (props.logPublishingOptions?.appLogEnabed) {
+      this.appLogGroup = new logs.LogGroup(this, 'AppLogs', {
+        logGroupName: `elasticsearch/domains/${props.domainName ?? id}/application-logs`,
```

Comment:
You can use the `this.domainName` Token instead of `props.domainName` and avoid the fallback.

Reply:
I think I am doing that now by setting `this.domainName` to `this.physicalName` and then referring to `this.domainName`. Is that what you meant?